src/udev/udev-rules.c

```diff
@@ -2022,6 +2022,12 @@ static int udev_rule_apply_token_to_event(
                         l = strpcpyl(&p, l, val, " ", NULL);
 
                 (void) udev_event_apply_format(event, token->value, p, l, false);
+                if (event->esc == ESCAPE_REPLACE) {
```

Comment:
hmm, if this is done for ENV{} assignments too now, I fiugre the man page explanation for string_escape= should be slightly updated.

it currently says:

>  Usually, control and other possibly unsafe characters are replaced in strings used for device naming. The mode of replacement can be specified with this option.

ENV{} doesn't really fit under the "strings used for device naming" umbrella. 

also, it should probably say that for the device naming stuff we default to escaping but for the env stuff we default to not escaping.